tests/foreman/cli/test_reporttemplates.py

```diff
@@ -313,139 +352,151 @@ def test_positive_generate_report_sanitized(self):
 
         :CaseImportance: Medium
         """
-
-    @tier2
-    @stubbed()
-    def test_negative_create_report_without_name(self):
-        """Try to create a report template with empty name
-
-        :id: 84b577db-144e-4771-a42e-e93887464986
-
-        :setup: User with reporting access rights
-
-        :steps:
-
-            1. hammer report-template create --name '' ...
-
-        :expectedresults: Report is not created
-
-        :CaseImportance: Medium
-        """
-
-    @tier2
-    @stubbed()
-    def test_negative_delete_locked_report(self):
-        """Try to delete a locked report template
-
-        :id: 84b577db-144e-4871-a42e-e93887464986
-
-        :setup: User with reporting access rights, some report template that is locked
-
-        :steps:
-
-            1. hammer report-template delete ...
-
-        :expectedresults: Report is not deleted
-
-        :CaseImportance: Medium
-        """
-
-    @tier2
+        os_name = gen_string('alpha') + "," + gen_string('alpha')
+        architecture = make_architecture()
+        partition_table = make_partition_table()
+        medium = make_medium()
+        os = make_os({
+            'name': os_name,
+            'architecture-ids': architecture['id'],
+            'medium-ids': medium['id'],
+            'partition-table-ids': partition_table['id'],
+        })
+
+        host_name = gen_string('alpha')
+        host = make_fake_host({
+            'name': host_name,
+            'architecture-id': architecture['id'],
+            'medium-id': medium['id'],
+            'operatingsystem-id': os['id'],
+            'partition-table-id': partition_table['id']
+        })
+
+        report_template = make_report_template({
+            'content': REPORT_TEMPLATE_FILE,
+        })
+
+        result = ReportTemplate.generate({
+            'name': report_template['name'],
+        })
+        self.assertIn(
+            host['name'] + "," + "\"" + host['operating-system']['operating-system'] + "\"",
```

Comment:
Explanation why there are quotation marks between host operating system is:
 when there is comma inside OS it has to be separated by quotes, in other case it will be interpreted as separator in csv not as OS name.